WordPress/Classes/ViewRelated/Comments/CommentsViewController.m

```diff
@@ -254,54 +254,50 @@ - (BOOL)tableView:(UITableView *)tableView canEditRowAtIndexPath:(NSIndexPath *)
-                                                                             title:NSLocalizedString(@"Unapprove", @"Unapproves a Comment")
-                                                                           handler:^(UITableViewRowAction * _Nonnull action, NSIndexPath * _Nonnull indexPath) {
-                                                                               [ReachabilityUtils onAvailableInternetConnectionDo:^{
-                                                                                   [weakSelf unapproveComment:comment];
-                                                                               }];
-                                                                           }];
+
+        // Unapprove Action
+        UIContextualAction *unapprove = [UIContextualAction contextualActionWithStyle:UIContextualActionStyleNormal title:NSLocalizedString(@"Unapprove", @"Unapproves a Comment") handler:^(UIContextualAction * _Nonnull action, __kindof UIView * _Nonnull sourceView, void (^ _Nonnull completionHandler)(BOOL)) {
+            [ReachabilityUtils onAvailableInternetConnectionDo:^{
+                [weakSelf unapproveComment:comment];
+            }];
+            completionHandler(YES);
+        }];
         
         unapprove.backgroundColor = [UIColor murielNeutral30];
         [actions addObject:unapprove];
     } else {
-        UITableViewRowAction *approve = [UITableViewRowAction rowActionWithStyle:UITableViewRowActionStyleNormal
-                                                                           title:NSLocalizedString(@"Approve", @"Approves a Comment")
-                                                                         handler:^(UITableViewRowAction * _Nonnull action, NSIndexPath * _Nonnull indexPath) {
-                                                                             [ReachabilityUtils onAvailableInternetConnectionDo:^{
-                                                                                 [weakSelf approveComment:comment];
-                                                                             }];
-                                                                         }];
+        // Approve Action
+        UIContextualAction *approve = [UIContextualAction contextualActionWithStyle:UIContextualActionStyleNormal title:NSLocalizedString(@"Approve", @"Approves a Comment") handler:^(UIContextualAction * _Nonnull action, __kindof UIView * _Nonnull sourceView, void (^ _Nonnull completionHandler)(BOOL)) {
+            [ReachabilityUtils onAvailableInternetConnectionDo:^{
+                [weakSelf approveComment:comment];
+            }];
+            completionHandler(YES);
+        }];
         
         approve.backgroundColor = [UIColor murielPrimary];
         [actions addObject:approve];
     }
     
-    return actions;
+    UISwipeActionsConfiguration *swipeActions = [UISwipeActionsConfiguration configurationWithActions:actions];
```

Comment:
Could we set `performsFirstActionWithFullSwipe` to `NO` on the `swipeActions`? This would prevent the user from accidentally trashing a comment by swiping too far or too quickly, and make it a more explicit action.